spec/units/extensions_spec.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'ostruct'
+
+describe Blueprinter::Extensions do
+  let(:all_extensions) {
+    [
+      foo_extension.new,
+      bar_extension.new,
+      zar_extension.new,
+    ]
+  }
+
+  let(:foo_extension) {
+    Class.new(Blueprinter::Extension) do
+      def pre_render(object, _blueprint, _view, _options)
+        obj = object.dup
+        obj.foo = "Foo"
+        obj
+      end
+    end
+  }
+
+  let(:bar_extension) {
+    Class.new(Blueprinter::Extension) do
+      def pre_render(object, _blueprint, _view, _options)
+        obj = object.dup
+        obj.bar = "Bar"
+        obj
+      end
+    end
+  }
+
+  let(:zar_extension) {
+    Class.new(Blueprinter::Extension) do
+      def self.something_else(object, _blueprint, _view, _options)
+        object
+      end
+    end
+  }
+
+  it 'should append extensions' do
+    extensions = Blueprinter::Extensions.new
+    extensions << foo_extension.new
+    extensions << bar_extension.new
+    extensions << zar_extension.new
+    expect(extensions.to_a.map(&:class)).to eq [
+      foo_extension,
+      bar_extension,
+      zar_extension,
+    ]
+  end
+
+  it "should initialize with extensions, removing any that don't have recognized extension methods" do
+    extensions = Blueprinter::Extensions.new(all_extensions)
+    expect(extensions.to_a.map(&:class)).to eq [
+      foo_extension,
+      bar_extension,
+      zar_extension,
+    ]
+  end
+
+  context '#pre_render' do
+    before :each do
+      Blueprinter.configure do |config|
+        config.extensions = all_extensions
+      end
+    end
+
+    after :each do
+      Blueprinter.configure do |config|
+      end
+    end
```

Comment:
what's this for? seems like a no-op

Reply:
It resets the config to defaults (i.e. no extensions).

Reply:
this is a hack then, aren't transactional_fixtures enabled? if they were, we wouldn't need this.

Reply:
Transactional fixtures? I don't believe that would affect in-memory BP configuration.

Reply:
yeah, my bad, I got mixed up with something else. 🤦 

Either ways, the snippet below from the `Configuration` class memoizes the `configuration` in memory through a class instance variable. So how exactly is `Blueprinter.configure {}` resetting the configs to defaults? It should simply return the memoized(in-memory) `configuration` right?
```ruby
  def self.configuration
    @configuration ||= Configuration.new
  end

  def self.configure
    yield configuration if block_given?
  end
```

You might want to have a look at [`SpecHelpers#reset_blueprinter_config`](https://github.com/search?q=repo%3Aprocore-oss/blueprinter%20reset_blueprinter_config!&type=code) instead - unless it's already being invoked.

Reply:
Oh, I see what you mean now. Fixing.